tests/gpflow/test_base.py

```diff
@@ -213,3 +214,38 @@ def test_construct_parameter_from_existing_parameter_value_becomes_invalid() ->
         gpflow.Parameter(initial_parameter, transform=transform)
 
     assert "gpflow.Parameter" in exc.value.message
+
+
+def test_construct_parameter_with_variable_shape() -> None:
+    parameter = gpflow.Parameter([[1, 2, 3]], shape=[None, None])
+
+    values = [
+        np.ones((0, 0), dtype=default_float()),
+        np.ones((1, 3), dtype=default_float()),
+        np.ones((3, 1), dtype=default_float()),
+        np.ones((3, 4), dtype=default_float()),
+    ]
+
+    for value in values:
+        parameter.assign(value)
+        np.testing.assert_equal(value, parameter.numpy())
+
+
+def test_construct_parameter_with_variable_shape__different_transformed_shape() -> None:
+    parameter = gpflow.Parameter(
+        [[1, 0], [2, 3]],
+        transform=triangular(),
+        pretransformed_shape=[None, None],
```

Comment:
Again, unless I'm misunderstanding, `[None, None]` is the _transformed_ shape (after applying `triangular()`) and `[None]` is the _pretransformed_ shape (which we get by applying `triangular().inverse`)?

Reply:
> Again, unless I'm misunderstanding, `[None, None]` is the _transformed_ shape (after applying `triangular()`) and `[None]` is the _pretransformed_ shape (which we get by applying `triangular().inverse`)?

Fixed.